libexec/bootstrap-scripts/main-deffile.sh

```diff
@@ -53,9 +53,23 @@ if [ -n "${BOOTSTRAP:-}" -a -z "${SINGULARITY_BUILDNOBASE:-}" ]; then
     fi
 fi
 
+# take a snapshot of the environment for later comparison
+if [ ${BOOTSTRAP} = "localimage" ]; then
+    export SINGULARITY_STARTING_ENVIRONMENT=$(eval_abort env -i ${SINGULARITY_libexecdir}/singularity/helpers/record-env.sh ${SINGULARITY_ROOTFS})
+    export SINGULARITY_STARTING_ENVSHA1=$(eval_abort sha1sum ${SINGULARITY_ROOTFS}/.singularity.d/env/*.sh | sha1sum)
+fi 
```

Comment:
So `$SINGULARITY_STARTING_ENVIRONMENT` is actually going to be the name of a temp file.  That temp file will contain the output of a `printenv` (basically the contents of the environment after sourcing everything in `/.singularity.d/env`.  And then yes `$SINGULARITY_STARTING_ENVSHA1` is just going to contain a hash.

I'm happy to export on a different line and give the vars default of empty, but I also don't know why.  😸